### PR TITLE
Make requirements more specific

### PR DESCRIPTION
**Pull request by gliptak**
Signed-off-by: Gábor Lipták <gliptak@gmail.com>

https://github.com/pydata/pandas-datareader/issues/694

### CODE REVIEW

File: requirements-doc.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+sphinx
+ipython
+matplotlib
+sphinx_rtd_theme
+doctr
```
Comment (addisonlynch) [inline requirements-doc.txt]: One small note here...is it necessary to add this to requirements-doc? Contribs don't need to install to work on documentation. It is only needed by Travis. Perhaps we could remove from here and install explicitly in CI

File: requirements-dev.txt [hunk 2]
```diff
@@ -1,6 +1,9 @@
-ipython
-requests_cache
-sphinx
-sphinx_rtd_theme
-black
-isort
+black; python_version > '3.5'
+flake8-bugbear; python_version > '3.5'
```
Comment (addisonlynch) [inline requirements-dev.txt]: We install ``flake8-bugbear`` in CI, but are we using it anywere?

File: requirements-doc.txt (hunk 1 above)
Comment (gliptak) [reply requirements-doc.txt]: I specifically separated to allow "text" handling of documentation build dependencies.

File: requirements-dev.txt (hunk 2 above)
Comment (gliptak) [reply requirements-dev.txt]: My understanding is that ```flake8-bugbear``` is a plugin into ```flake8```, so it is a development dependency

Comment (bashtage) [reply requirements-dev.txt]: bugbear is optional and can't be a dependency since it doesn't support Python 2.7.  It can go in after this release.

Comment (bashtage) [reply requirements-dev.txt]: The flake8 'B' tests are bugbear.

Comment (gliptak) [reply requirements-dev.txt]: the specific syntax above should only install it for ```3.5+```

File: docs/requirements.txt [hunk 3]
```diff
@@ -1,7 +0,0 @@
-lxml
```
Comment (bashtage) [inline docs/requirements.txt]: This file has to remain.  It is used by readthedocs.  This should have all docbuild requirement except doctr.

File: requirements-dev.txt (hunk 2 above)
Comment (bashtage) [reply requirements-dev.txt]: Ignore my previous.  This method is fine.

File: requirements-doc.txt (hunk 1 above)
Comment (bashtage) [reply requirements-doc.txt]: This file should go into docs/requirements for RTD compat.

**Comment by bashtage** [comment]
One more: doctr needs to be pip installed the if statement

File: docs/requirements.txt (hunk 3 above)
Comment (gliptak) [reply docs/requirements.txt]: thx. I didn't know RTD uses this. I will update

**Comment by gliptak** [comment]
> 
> 
> One more: doctr needs to be pip installed the if statement

So ```doctr``` needs a minimum Python version?

**Comment by bashtage** [comment]
> So `doctr` needs a minimum Python version?

No, it just isn't a docbuild requirement. It is travis specific, so should only be installed if one travis and in docbuild.

**Comment by bashtage** [comment]
Thanks.

**Pull request merged**